src/System.Private.CoreLib/shared/System/Diagnostics/DebugProvider.cs

```diff
@@ -73,19 +73,31 @@ public virtual void WriteLine(string? message)
         private sealed class DebugAssertException : Exception
         {
             internal DebugAssertException(string? stackTrace) :
-                base(Environment.NewLine + stackTrace)
+                base(stackTrace)
             {
             }
 
             internal DebugAssertException(string? message, string? stackTrace) :
-                base(message + Environment.NewLine + Environment.NewLine + stackTrace)
+                base(Terminate(message) + stackTrace)
             {
             }
 
             internal DebugAssertException(string? message, string? detailMessage, string? stackTrace) :
-                base(message + Environment.NewLine + detailMessage + Environment.NewLine + Environment.NewLine + stackTrace)
+                base(Terminate(message) + Terminate(detailMessage) + stackTrace)
```

Comment:
This was causing the extra newline when the message was empty, as pointed out in feedback.